File: python/change-printer-message.py
#!/usr/bin/env python
import socket
import sys

if len(sys.argv) != 3:
	print("ERROR: Need to specify both a printer address and a message")
	sys.exit()

s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect((sys.argv[1], 9100))
s.send("\x1B%-12345X@PJL RDYMSG DISPLAY = \"" + sys.argv[2] + "\"\r\n\x1B%-12345X\r\n")
s.close()

print("Message successfully sent to printer")

File: ruby/change-printer-message.rb
#!/usr/bin/env ruby
require 'socket'

abort "ERROR: Need to specify both a printer address and a message" unless ARGV.length == 2

s = TCPSocket.open(ARGV[0], 9100)
s.puts("\x1B%-12345X@PJL RDYMSG DISPLAY = \"" + ARGV[1] + "\"\r\n\x1B%-12345X\r\n")
s.close

puts "Message successfully sent to printer"
